### ISSUE TITLE
Update with PSA Crypto 1.0.0 interface

### ISSUE THREAD

**Issue by hug-dev** [parallaxsecond/parsec#129]
Parsec service needs to be updated to use parallaxsecond/parsec-interface-rs#4.

The new interface introduced the `alg` parameter for sign/verify operations. The algorithm used for signing/verifying needs to be checked against the key policy (which has to be stored somewhere).

**Comment by ionut-arm** [comment]
If the underlying API doesn't offer the option to retrieve said algorithm, we can probably repurpose the `KeyIDManager` to cover more key-related information (and potentially change it to `KeyInfoManager`)

**Comment by hug-dev** [comment]
About modification in the Key ID Manager and checking for policy I am thinking of the following things to do:
* add integration tests, per provider, checking that `PsaErrorNotPermitted` is returned if:
  * the algorithm is not compatible with the type of the key
  * the operation is not permitted by the usage flags
  * the algorithm is not the permitted algorithm in the policy
* modify the Key ID Manager trait and implementors to use a `KeyInfo` structure (instead of just the key ID):
```rust
/// Information stored about a key
#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
pub struct KeyInfo {
    /// Reference to a key in the Provider
    pub id: Vec<u8>,
    /// Attributes of a key
    pub attributes: KeyAttributes,
}
```
* for each operation, providers need to check if it is permitted and return `PsaErrorNotPermitted` if not. For that implements the following method on the `KeyAttributes` structure:
```rust
fn permits(&self, op: NativeOperation) -> bool
```
to check if the key permits the operation, ie the conditions on the first point are respected. That way, providers get the key attributes from the KIM, directly call this method and return `PsaErrorNotPermitted` if it's `false`.

**Comment by hug-dev** [comment]
I will also add here two things to check as part of this task:

- [ ] `ResponseStatus`: check when it is adequate to return service ones or PSA ones
- [ ] For each operation: check if the behaviour is as described in the API and update the book

**Comment by hug-dev** [comment]
Tests have been merged in #135

**Comment by hug-dev** [comment]
All updates are done! [Last PR](https://github.com/parallaxsecond/parsec/pull/144)
